python/cuml/dask/ensemble/randomforestcommon.py

```diff
@@ -0,0 +1,173 @@
+            sd = self.n_estimators_per_worker[i-1] + seeds[i-1]
+            seeds.append(sd)
+
+        key = str(uuid1())
+        self.rfs = {
+            worker: self.client.submit(
+                model_func,
+                n_estimators=self.n_estimators_per_worker[n],
+                seed=seeds[n],
+                **kwargs,
+                key="%s-%s" % (key, n),
+                workers=[worker],
+            )
+            for n, worker in enumerate(self.workers)
+        }
+
+        rfs_wait = list()
+        for r in self.rfs.values():
+            rfs_wait.append(r)
+
+        wait(rfs_wait)
+        raise_exception_from_futures(rfs_wait)
+
+    def _fit(self, model, dataset, convert_dtype):
+        data = DistributedDataHandler.create(dataset, client=self.client)
+        self.datatype = data.datatype
+        futures = list()
+        for idx, (worker, worker_data) in \
+                enumerate(data.worker_to_parts.items()):
+            futures.append(
+                self.client.submit(
+                    _func_fit,
+                    model[worker],
+                    worker_data,
+                    convert_dtype,
+                    workers=[worker],
+                    pure=False)
+            )
+        wait(futures)
+        raise_exception_from_futures(futures)
+        return self
+
+    def _concat_treelite_models(self):
+        """
+        Convert the cuML Random Forest model present in different workers to
+        the treelite format and then concatenate the different treelite models
+        to create a single model. The concatenated model is then converted to
+        bytes format.
+        """
+        mod_bytes = []
+        for w in self.workers:
+            mod_bytes.append(self.rfs[w].result().model_pbuf_bytes)
+        last_worker = w
+        all_tl_mod_handles = []
+        model = self.rfs[last_worker].result()
+        for n in range(len(self.workers)):
+            all_tl_mod_handles.append(model._tl_model_handles(mod_bytes[n]))
+        concat_model_handle = model._concatenate_treelite_handle(
+            treelite_handle=all_tl_mod_handles)
+
+        model._concatenate_model_bytes(concat_model_handle)
+        return model
+
+    def _predict_using_fil(self, X, delayed, **kwargs):
+        data = DistributedDataHandler.create(X, client=self.client)
+        self.datatype = data.datatype
+        return self._predict(X, delayed=delayed, **kwargs)
+
+    def _get_params(self, deep):
+        model_params = list()
+        for n, worker in enumerate(self.workers):
+            model_params.append(
+                self.client.submit(
+                    _func_get_params,
+                    self.rfs[worker],
+                    deep,
+                    workers=[worker]
+                )
+            )
+        wait(model_params)
+        raise_exception_from_futures(model_params)
+        params_of_each_model = list()

```

Comment:
are lines 116-118 equivalent to:

`[params.result() for params in model_params]`? I think that comprehension would be clearer and more concise.
As a general tip, if you're iterating over a list in python using integer indices, consider whether there is an alternative iteration strategy that goes directly to the objects.